core/src/main/java/io/fabric8/maven/core/util/kubernetes/KubernetesResourceUtil.java

```diff
@@ -387,7 +387,7 @@ private static void addIfNotExistent(Map<String, Object> fragment, String key, S
             Map<String, Object> ret = mapper.readValue(file, typeRef);
             return ret != null ? ret : new HashMap<String, Object>();
         } catch (JsonProcessingException e) {
-            throw new JsonMappingException(String.format("[%s] %s", file, e.getMessage()), e.getLocation(), e);
+            throw new JsonProcessingException(String.format("[%s] %s", file, e.getMessage()), e.getLocation(), e) {};
```

Comment:
just curious. how do curly braces help here? I see that you're trying to invoke a protected constructor. Does the curly brace turn it into an anonymous class object extending the `JsonProcessingException` (just a wild guess!). I'm learning Java :)

Reply:
Yes, the curely braces creates a anonymous class extending `JsonProcessingException`. I chose this solution for two reasons:
1. The deprecated constructor on `JsonMappingException` suggests a replacement using a factory with `JsonParser` as one of it's parameters. I didn't figure out how to get access to it from the `ObjectMapper`.
1. The purpose of this catch block is just to enrich the error message presented to the user with the source file information. So I thought it was best not to transform a generic exception (`JsonProcessingException`) into a more specific (`JsonMappingException`). It could end up being wrong/misleading.

Reply:
A protected constructor is not nessacarily considered part of an API, and I think Jackson fairly can make this constructor package-protected (or even remove it) at some point - without a new major version, ref. SemVer. I tend to use just plain `RuntimeException` to wrap exceptions like this one. I am not a big fan of checked exceptions in general....